### ISSUE TITLE
Adaptive card with controls not readable to screen reader

### ISSUE THREAD

**Issue by venkatx5** [microsoft/BotFramework-WebChat#3868]
We are using webchat-es5.js for our Virtual assistant and some of the customers are using screen reader to access it. The screen reads the normal card and user messages properly. But for Adaptive card with controls, it doesn't read the content. 

## Screenshots
![image](https://user-images.githubusercontent.com/11777507/116230046-cea69280-a774-11eb-85e0-b9bd6740e082.png)

![image](https://user-images.githubusercontent.com/11777507/116229073-b2562600-a773-11eb-8f93-20a110a9f40d.png)

## Version
webchat es5 version 4.12.0
https://cdn.botframework.com/botframework-webchat/4.12.0/webchat-es5.js
Integrating the CDN version js directly on the page using script tag (No Iframe). Desktop Chrome Browser.

```
<meta name="botframework-directlinespeech:version" content="4.12.0">
<meta name="botframework-webchat:bundle:variant" content="full-es5">
<meta name="botframework-webchat:bundle:version" content="4.12.0">
<meta name="botframework-webchat:core:version" content="4.12.0">
<meta name="botframework-webchat:ui:version" content="4.12.0">
```

## Describe the bug
We are using webchat-es5.js for our Virtual assistant and some of the customers are using screen reader to access it. The screen reader reads normal card text and user input text. But if Bot response is adaptive card with controls, then it doesn't read the card content. 
•	On a Virtual Agent message/response with no links or click-able buttons, assistive technology like screen readers can read aloud the full message when in focus. Please see the code example below:
**Refer Screenshot 1 above**
1) A typical message in the Virtual Agent (no links/buttons) is arranged in an unordered list < ul > of messages. One message/response is one list item < li >.
2) The message to be spoken is within the < article > tag which contains paragraph tags < p > with one of them carrying the actual Virtual Agent response.
3) The “aria-labelledby” attribute in the list item tag < li > for the message is looking at the “id” attribute of the < article > tag for the message. This programmatically associates the Virtual Agent message with the control.
This approach is working normally and there is no issue. However when the Virtual Agent adds links or buttons to a response, things do not work as expected…

**Refer Screenshot 2 above**
 
• The “aria-labelledby” attribute for the list item (for the message) is still pointing to the “id” within the < article > tag, but the only information within that that tag is:
o	“Bot CB said:”
o	“1 attachment.”
o	The date stamp
o	and “Press ENTER to interact”
When we use the mouse or keyboard to focus that list item for the message, the 4 sub-items mentioned mentioned are the only information available to assistive technology. The paragraph with the actual Virtual Agent response is much lower in the DOM (visible in the previous example) and since it isn’t programmatically associated with the list item via “aria-labelledby” or any equivalent method like in the first example, it isn’t available to be read from assistive technology like screen readers. Low vision or other users relying a screen reader will not know what to focus and interact with as currently configured. This MUST be corrected.

## Steps to reproduce
1. Deploy Web App Bot with adaptive cards.
2. Start NVDA screen reader or JAWS 2021 screen reader.
3. Open the web app bot on browser.
4. Interact with the bot till the adaptive card response arrives.
5. Observe how the screen reader reads the bot response.

## Expected behavior
Using ARIA to programmatically associate the Virtual Agent response text with the list item for the message or moving the message text into the <article> tag itself somewhere so it is available to assistive technology like screen readers will correct the issue.

Please advice if any other fix to make the screen reader to read the adaptive card content.
[Bug]

**Comment by corinagum** [comment]
Hi, Thanks for filing. 
1. Just in case, could you try this behavior on Web Chat 4.13.0? I anticipate the behavior won't change, but let's be sure. 
2. Could you provide the JSON for the Adaptive Card that is being used?

The topics are slightly different, but I believe the fix will be related to the changes for https://github.com/microsoft/BotFramework-WebChat/issues/3804

**Comment by venkatx5** [comment]
The issue remains even after change the webchat version to 4.13.0.
Will share the Adaptive card json.

**Comment by venkatx5** [comment]
Adaptive card JSON.

{
  "type": "message",
  "serviceUrl": "http://localhost:56223",
  "channelId": "emulator",
  "from": {
    "id": "db8f84e0-b217-11eb-b23******** * * *89",
    "name": "Bot",
    "role": "bot"
  },
  "conversation": {
    "id": "4dff8ba0-b246-11**************"
  },
  "recipient": {
    "id": "ffb569b4-b872-4ab2**************",
    "role": "user"
  },
  "locale": "en-US",
  "inputHint": "expectingInput",
  "attachments": [
    {
      "contentType": "application/vnd.microsoft.card.adaptive",
      "content": {
        "type": "AdaptiveCard",
        "version": "1.0",
        "body": [
          {
            "type": "TextBlock",
            "id": "firstresponse",
            "text": "Benefits are paid to a debit card provided by the state's vendor, ABCD Inc. The card will be issued to you at the time your first eligible week processes. It normally takes between 7-10 days to receive the card. Do not throw the debit card away. You have two (2)..."
          },
          {
            "type": "TextBlock",
            "id": "secondresponse",
            "text": "Benefits are paid to a debit card provided by the state's vendor, ABCD Inc. The card will be issued to you at the time your first eligible week processes. It normally takes between 7-10 days to receive the card. Do not throw the debit card away. You have two (2) options to receive payments. You may receive your payments via the debit card or you may transfer the payments from the debit card into your own U.S. banking account, for no fee. To receive payments to the debit card, you must activate the card by following the instructions sent with the card. To set up this recurring direct transfer, you must activate your card and then visit [website.com](**********) or call *********** and use the automated system to enter your banking information. If you have previously been issued a debit card for payment within the past three (3) years, your payment will be issued to that debit card unless otherwise specified. If you no longer have your previously issued debit card, please contact ABCD at ******* * * *.",
            "isVisible": false
          },
          {
            "id": "showmorebtns",
            "type": "ActionSet",
            "actions": [
              {
                "type": "Action.ToggleVisibility",
                "targetElements": [
                  {
                    "elementId": "firstresponse",
                    "isVisible": false
                  },
                  {
                    "elementId": "secondresponse",
                    "isVisible": true
                  },
                  {
                    "elementId": "showmorebtns",
                    "isVisible": false
                  },
                  {
                    "elementId": "hidemorebtns",
                    "isVisible": true
                  }
                ],
                "title": "Read more"
              }
            ]
          },
          {
            "id": "hidemorebtns",
            "type": "ActionSet",
            "actions": [
              {
                "type": "Action.ToggleVisibility",
                "targetElements": [
                  {
                    "elementId": "secondresponse",
                    "isVisible": false
                  },
                  {
                    "elementId": "showmorebtns",
                    "isVisible": true
                  },
                  {
                    "elementId": "firstresponse",
                    "isVisible": true
                  },
                  {
                    "elementId": "hidemorebtns",
                    "isVisible": false
                  }
                ],
                "title": "Read less"
              }
            ],
            "isVisible": false
          },
          {
            "type": "TextBlock",
            "text": "Does this answer your question?"
          }
        ],
        "actions": [
          {
            "type": "Action.Submit",
            "id": "Success",
            "data": {
              "IsSatisfied": "true",
              "UserAskedQuestion": "How and when are payments made",
              "ResolvedIntent": "payment_debit_card_general"
            },
            "iconUrl": "*************************** * * * "
          },
          {
            "type": "Action.Submit",
            "id": "Failure",
            "data": {
              "IsSatisfied": "false",
              "UserAskedQuestion": "How and when are payments made",
              "ResolvedIntent": "payment_debit_card_general"
            },
            "iconUrl": " * * ************************* * * *"
          },
          {
            "type": "Action.ShowCard",
            "card": {
              "type": "AdaptiveCard",
              "version": "1.0",
              **"speak": "benefits are paid to a debit card provided by the state's vendor, ABCD Inc. The card will be issued to you at the time your first eligible week processes. It normally takes between 7-10 days to receive the card. Do not throw the debit card away. You have two (2) options to receive payments. You may receive your payments via the debit card or you may transfer the payments from the debit card into your own banking account, for no fee. To receive payments to the debit card, you must activate the card by following the instructions sent with the card. To set up this recurring direct transfer, you must activate your card and then visit [website.com](******************************) or call ****************************** and use the automated system to enter your banking information. If you have previously been issued a debit card for payment within the past three (3) years, your payment will be issued to that debit card unless otherwise specified. If you no longer have your previously issued debit card, please contact ABCD at **************************** * * .",**
              "actions": [
                {
                  "type": "Action.Submit",
                  "id": "Do I need to wait to file until after my severance, accrued vacation pay, or payments are made?",
                  "data": "Do I need to wait to file until after my severance, accrued vacation pay, or payments are made?",
                  "title": "Do I need to wait to file until after my severance, accrued vacation pay, or payments are made?"
                },
                {
                  "type": "Action.Submit",
                  "id": "When are benefits paid?",
                  "data": "When are benefits paid?",
                  "title": "When are benefits paid?"
                }
              ]
            },
            "title": "View Recommended Help"
          },
          {
            "type": "Action.Submit",
            "id": "return",
            "data": "Return to main menu",
            "title": "Return to main menu"
          }
        ]
      }
    }
  ],
  "replyToId": "815ef170-b2 * * * *********************** * * * *e",
  "id": "868a7d90-b* * ************************** * * 9ee29c2ede",
  "localTimestamp": "2021-05-11T16:19:13+05:30",
  "timestamp": "2021-05-11T10:49:13.961Z"
}

**Comment by corinagum** [comment]
A few notes: 
- When I paste your card into the [Adaptive Card Designer](https://adaptivecards.io/designer/), I get some validation issues. Your card version is notated as version 1.0, but you are using 1.2+ schema properties, which may cause rendering errors
- In order for the card to be read out when first added to the transcript, you will need to add the `speak` property to the root of the card. That `speak` property will be read out, and then on `'Enter'` interaction, the input controls will be spoken via tab navigation. 
    - I see you have the `speak` property commented out in one of the `TextBlocks`, but it will need to be in the [root of the card (same level as `body`)](https://adaptivecards.io/explorer/AdaptiveCard.html)

Using the speak property for accessibility is Web Chat's work-around for having concise 'summary' text of the Adaptive Card to be read by AT. 

Please note that the Adaptive Cards Designer is a good place to validate your cards, taking Web Chat out of the equation to determine what kind of bug you are experiencing.

Closing this issue as resolved.
